src/shared/usbspec.rs

```diff
@@ -0,0 +1,952 @@
+STRUCT!{#[repr(packed)] struct USB_BOS_DESCRIPTOR {
+    bLength: UCHAR,
+    bDescriptorType: UCHAR,
+    wTotalLength: USHORT,
+    bNumDeviceCaps: UCHAR,
+}}
+pub type PUSB_BOS_DESCRIPTOR = *mut USB_BOS_DESCRIPTOR;
+pub const USB_DEVICE_CAPABILITY_WIRELESS_USB: UCHAR = 0x01;
+pub const USB_DEVICE_CAPABILITY_USB20_EXTENSION: UCHAR = 0x02;
+pub const USB_DEVICE_CAPABILITY_SUPERSPEED_USB: UCHAR = 0x03;
+pub const USB_DEVICE_CAPABILITY_CONTAINER_ID: UCHAR = 0x04;
+pub const USB_DEVICE_CAPABILITY_PLATFORM: UCHAR = 0x05;
+pub const USB_DEVICE_CAPABILITY_POWER_DELIVERY: UCHAR = 0x06;
+pub const USB_DEVICE_CAPABILITY_BATTERY_INFO: UCHAR = 0x07;
+pub const USB_DEVICE_CAPABILITY_PD_CONSUMER_PORT: UCHAR = 0x08;
+pub const USB_DEVICE_CAPABILITY_PD_PROVIDER_PORT: UCHAR = 0x09;
+pub const USB_DEVICE_CAPABILITY_SUPERSPEEDPLUS_USB: UCHAR = 0x0A;
+pub const USB_DEVICE_CAPABILITY_PRECISION_TIME_MEASUREMENT: UCHAR = 0x0B;
+pub const USB_DEVICE_CAPABILITY_BILLBOARD: UCHAR = 0x0D;
+pub const USB_DEVICE_CAPABILITY_CONFIGURATION_SUMMARY: UCHAR = 0x10;
+STRUCT!{#[repr(packed)] struct USB_DEVICE_CAPABILITY_USB20_EXTENSION_DESCRIPTOR_bmAttributes {
+    AsUlong: ULONG,
+}}
+BITFIELD!{USB_DEVICE_CAPABILITY_USB20_EXTENSION_DESCRIPTOR_bmAttributes AsUlong: ULONG [
+    Reserved set_Reserved[0..1],
+    LPMCapable set_LPMCapable[1..2],
+    BESLAndAlternateHIRDSupported set_BESLAndAlternateHIRDSupported[2..3],
+    BaselineBESLValid set_BaselineBESLValid[3..4],
+    DeepBESLValid set_DeepBESLValid[4..5],
+    Reserved1 set_Reserved1[5..8],
+    BaselineBESL set_BaselineBESL[8..12],
+    DeepBESL set_DeepBESL[12..16],
+    Reserved2 set_Reserved2[16..32],
+]}
+STRUCT!{#[repr(packed)] struct USB_DEVICE_CAPABILITY_USB20_EXTENSION_DESCRIPTOR {
+    bLength: UCHAR,
+    bDescriptorType: UCHAR,
+    bDevCapabilityType: UCHAR,
+    bmAttributes: USB_DEVICE_CAPABILITY_USB20_EXTENSION_DESCRIPTOR_bmAttributes,
+}}
+pub type PUSB_DEVICE_CAPABILITY_USB20_EXTENSION_DESCRIPTOR
+    = *mut USB_DEVICE_CAPABILITY_USB20_EXTENSION_DESCRIPTOR;
+pub const USB_DEVICE_CAPABILITY_USB20_EXTENSION_BMATTRIBUTES_RESERVED_MASK: ULONG = 0xFFFF00E1;
+STRUCT!{struct USB_DEVICE_CAPABILITY_POWER_DELIVERY_DESCRIPTOR_bmAttributes_s {
+    BitField: ULONG,
+}}
+BITFIELD!{USB_DEVICE_CAPABILITY_POWER_DELIVERY_DESCRIPTOR_bmAttributes_s BitField: ULONG [
+    Reserved set_Reserved[0..1],
+    BatteryCharging set_BatteryCharging[1..2],
+    USBPowerDelivery set_USBPowerDelivery[2..3],
+    Provider set_Provider[3..4],
+    Consumer set_Consumer[4..5],
+    ChargingPolicy set_ChargingPolicy[5..6],
+    TypeCCurrent set_TypeCCurrent[6..7],
+    Reserved2 set_Reserved2[7..8],
+    ACSupply set_ACSupply[8..9],
+    Battery set_Battery[9..10],
+    Other set_Other[10..11],
+    NumBatteries set_NumBatteries[11..14],
+    UsesVbus set_UsesVbus[14..15],
+    Reserved3 set_Reserved3[15..32],
+]}
+UNION2!{#[repr(packed)] union USB_DEVICE_CAPABILITY_POWER_DELIVERY_DESCRIPTOR_bmAttributes {
```

Comment:
This should be a single field struct with the bitfield applied directly to `AsUlong`.